### DISCUSSION TITLE
Including descriptions from schema in the generated types for documents

### DISCUSSION THREAD

**Discussion by sam-outschool** [dotansimha/graphql-code-generator#8239] [Ideas]
In my project, we are looking to document our Graph more thoroughly using [descriptions](https://spec.graphql.org/draft/#sec-Descriptions). Using `graphql-codegen` means that when we have descriptions for a field, they're also included in the types as a jsdoc, available for use by code editors on hover.

This behavior is perfect when working with types on our server, and I'd like to have something similar for the client side as well. I'd like to propose an option to include schema descriptions for the document-generated types as well. There's a chance I'm missing something and that this is ill-advised, but I don't see a situation where we would want to have separate descriptions for the schema and documents.

**Comment by charlypoly** [comment]
Hi @sam-outschool!
I don't see any part of GraphQL Code Generator code that handles adding Schema fields' comments to Operations fields.

**Reply by sam-outschool** [reply]
Yeah, I expect supporting a new option like this would require some code changes. Do you mean that there isn't a good place in the code to support this option? I was hoping that there was a pass over the operations fields that gets the relevant type from the schema, and that we could include comments during that pass. I'm happy to give the code change a shot with some direction, assuming it's possible!

**Reply by charlypoly** [reply]
I guess this change would apply to the `typescript-operations` plugin, yes 👀 
Feel free to open a PR and ping me, I'll help you along the way!
